media-proxy/src/mesh/conn_rdma_rx.cc

```diff
@@ -0,0 +1,149 @@
+#include "conn_rdma_rx.h"
+#include <stdexcept>
+#include <queue>
+
+namespace mesh::connection {
+
+RdmaRx::RdmaRx() : Rdma() {
+    _kind = Kind::receiver; // Set the Kind in the constructor
+    dir = direction::RX;    // Set the direction in the constructor
+}
+
+RdmaRx::~RdmaRx()
+{
+    // Any specific cleanup for Rx can go here
+}
+
+Result RdmaRx::configure(context::Context& ctx, const mcm_conn_param& request,
+                         const std::string& dev_port, libfabric_ctx *& dev_handle)
+{
+    return Rdma::configure(ctx, request, dev_port, dev_handle);
+}
+
+Result RdmaRx::start_threads(context::Context& ctx)
+{
+
+    process_buffers_thread_ctx = context::WithCancel(ctx);
+    rdma_cq_thread_ctx = context::WithCancel(ctx);
+
+    try {
+        handle_process_buffers_thread = std::jthread(
+            [this]() { this->process_buffers_thread(this->process_buffers_thread_ctx); });
+    } catch (const std::system_error& e) {
+        log::fatal("Failed to start thread")("error", e.what());
+        return Result::error_thread_creation_failed;
+    }
+
+    try {
+        handle_rdma_cq_thread =
+            std::jthread([this]() { this->rdma_cq_thread(this->rdma_cq_thread_ctx); });
+    } catch (const std::system_error& e) {
+        log::fatal("Failed to start thread")("error", e.what());
+        return Result::error_thread_creation_failed;
+    }
+
+    return Result::success;
+}
+
+/**
+ * @brief Handles the buffer processing logic for RDMA in a dedicated thread.
+ * 
+ * Continuously consumes available buffers from the queue and prepares them for RDMA reception
+ * by passing them to the RDMA endpoint. If no buffers are available, the thread waits for 
+ * notification of buffer availability. Ensures graceful handling of errors and context cancellation.
+ * 
+ * @param ctx The context for managing thread cancellation and operations.
+ */
+void RdmaRx::process_buffers_thread(context::Context& ctx)
+{
+
+    while (!ctx.cancelled()) {
+        void *buf = nullptr;
+
+        // Process all available buffers in the queue
+        while (!ctx.cancelled()) {
+            Result res = consume_from_queue(ctx, &buf);
+            if (res == Result::success && buf != nullptr) {
+                int err = libfabric_ep_ops.ep_recv_buf(ep_ctx, buf, trx_sz, buf);
+                if (err) {
+                    log::error("Failed to pass empty buffer to RDMA to receive into")
+                              ("buffer_address", buf)("error", fi_strerror(-err))
+                              (" ",kind2str(_kind));
+                    add_to_queue(buf);
```

Comment:
add_to_queue() returns a return value, check it.

Reply:
fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/commit/737483c86a4bf88fdd86b9e223e58acf7f3b7d6d